File: project/module-core/src/main/java/cn/fd/ratziel/function/exception/ArgumentNotFoundException.java
package cn.fd.ratziel.function.exception;

/**
 * ArgumentNotFoundException
 *
 * @author TheFloodDragon
 * @since 2024/8/16 18:28
 */
public final class ArgumentNotFoundException extends Exception {

    public ArgumentNotFoundException(Class<?> type) {
        super("Cannot found the argument: " + type.getName() + " !");
    }

}

File: project/module-core/src/main/kotlin/cn/fd/ratziel/function/argument/exception/ArgumentNotFoundException.kt
package cn.fd.ratziel.function.argument.exception

/**
 * ArgumentNotFoundException
 *
 * @author TheFloodDragon
 * @since 2024/5/1 14:12
 */
class ArgumentNotFoundException : ArgumentException {

    constructor(type: Class<*>, throwable: Throwable) : super("Cannot found the argument: ${type.name} !", throwable)

    constructor(type: Class<*>) : super("Cannot found the argument: ${type.name} !")

    constructor() : super("Cannot found the argument!")

}